Lib/test/test_os.py

```diff
@@ -3122,6 +3122,19 @@ def test_stty_match(self):
         self.assertEqual(expected, actual)
 
 
+@unittest.skipUnless(hasattr(os, 'memfd_create'), 'requires os.memfd_create')
+class MemfdCreateTests(unittest.TestCase):
+    def test_memfd_create(self):
+        fd = os.memfd_create("Hi", os.MFD_CLOEXEC)
```

Comment:
This may leak an fd if one of the checks is failing. Please make sure that the fd is ``os.close()``-ed on error.

Reply:
Right. I added `self.addCleanup(os.close, fd)` after the `self.assertNotEqual(fd, -1)` in order to avoid calling `os.close()` on `-1` in the unlikely event of a failure there (as the `self.assertNotEqual(fd, -1)` should never fail at that point).

Reply:
(I was just posting my own thought process -- I don't have a lot of experience with the unit testing of CPython and CPython internals).